Sources/PackageGraph/PackageGraph.swift

```diff
@@ -35,7 +35,8 @@ enum PackageGraphError: Swift.Error {
         targetName: String,
         packageIdentifier: String
     )
-
+    /// Dependency between a given target of its type and a dependent target/product of its type is unsupported
+    case unsupported(targetName: String, targetType: String, dependencyName: String, dependencyType: String,  dependencyPackage: String?)
```

Comment:
unsupportedDependency?

Reply:
It's intended to be generalized so it can handle other unsupported cases if needed; lmk if you think otherwise.

Reply:
generally we keep them specific, as it gives us more flexibility over error messages